### PR TITLE
[WIP]Model extends issue

### PR DESCRIPTION
**Pull request by skywing918**
Template Instance should not inline if response

### CODE REVIEW

File: packages/openapi3/src/schema-emitter.ts [hunk 1]
```diff
@@ -840,7 +842,30 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
       };
     }
 
-    if (shouldInline(this.emitter.getProgram(), type)) {
+    const checkNeedForceRef = (
```
Comment (wanlwanl) [inline packages/openapi3/src/schema-emitter.ts:845]: should be part of `shouldInline`?

File: .chronus/changes/ModelExtendsIssue-2024-9-16-14-34-22.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+changeKind: fix
+packages:
+  - "@typespec/openapi3"
+---
+
+template instance should not inline if it is response
```
Comment (wanlwanl) [inline .chronus/changes/ModelExtendsIssue-2024-9-16-14-34-22.md:7]: parameter also has the same behavior no?

File: packages/openapi3/src/schema-emitter.ts [hunk 3]
```diff
@@ -840,7 +842,30 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
...
+      schema: ObjectBuilder<any>,
+    ): boolean => {
+      if (getFriendlyName(program, type)) {
+        return false;
+      }
+      if (type.kind === "Model" && isTemplateInstance(type)) {
+        return Array.from(type.properties.values()).some((prop) => {
+          if (prop.type?.kind === "Model") {
+            const curr = schema.properties[prop.name];
+            if (isArrayModelType(program, prop.type)) {
```
Comment (wanlwanl) [inline packages/openapi3/src/schema-emitter.ts:857]: Why limit to array?

File: packages/openapi3/test/template-instance.test.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    model Pet {
+      choice_text: string;
+    }
+    ${model}
+    @post op create(prompt: string): ${modelName};
+    `);
+  ok(openApi.components.schemas.Cat, "expected definition named Cat");
+  ok(openApi.components.schemas.Pet, "expected definition named Pet");
+  deepStrictEqual(openApi.paths["/"].post.responses["200"].content["application/json"].schema, {
+    $ref: "#/components/schemas/Cat",
```
Comment (wanlwanl) [inline packages/openapi3/test/template-instance.test.ts:19]: does Cat exists? consider add a check?

File: packages/openapi3/src/schema-emitter.ts (hunk 1 above)
Comment (skywing918) [reply packages/openapi3/src/schema-emitter.ts:845]: no, the scenario is very special,it is only handled when `array of model` and `response body`

File: .chronus/changes/ModelExtendsIssue-2024-9-16-14-34-22.md (hunk 2 above)
Comment (skywing918) [reply .chronus/changes/ModelExtendsIssue-2024-9-16-14-34-22.md:7]: no

**Comment by wanlwanl** [comment]
current behavior is current. the improvement is in pre-design stage

**Pull request closed without merging**